hugr-model/src/v0/mod.rs

```diff
@@ -132,29 +133,39 @@ macro_rules! define_index {
 }
 
 define_index! {
-    /// Index of a node in a hugr graph.
+    /// Id of a node in a hugr graph.
     #[derive(Debug, derive_more::Display, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
     pub struct NodeId(pub u32);
 }
 
 define_index! {
     /// Index of a link in a hugr graph.
     #[derive(Debug, derive_more::Display, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
-    pub struct LinkId(pub u32);
+    pub struct LinkIndex(pub u32);
 }
 
 define_index! {
-    /// Index of a region in a hugr graph.
+    /// Id of a region in a hugr graph.
     #[derive(Debug, derive_more::Display, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
     pub struct RegionId(pub u32);
 }
 
 define_index! {
-    /// Index of a term in a hugr graph.
+    /// Id of a term in a hugr graph.
     #[derive(Debug, derive_more::Display, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
     pub struct TermId(pub u32);
 }
 
+/// The id of a link consisting of its region and the link index.
+#[derive(Debug, derive_more::Display, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[display("{_0}#{_1}")]
+pub struct LinkId(pub RegionId, pub LinkIndex);
+
+/// The id of a variable consisting of its node and the variable index.
```

Comment:
Might be overkill but consider ultra-explicitness "The id of a variable being the node declaring the variable and the index within the variables declared by that node"